collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,230 @@
+# Extend Kyma CLI
+
+Kyma cli has been used to provision a cluster and install kyma into. We should look into extend the current functionality set such that we can provide functionality to access various features of kyma.
+
+The extension of functionality would enable people with no or less kubernetes knowledge can use kyma without any issue.
+
+## Security
+
+Kyma cli should support RBAC. This would be realised by downloading the requisite kubeconfig file. The admins should be able to use more infrastructure commands and developers should be able to use the kyma cluster with restricted rights.
+
+## Architecture
+
+  We should follow the modular approach with each component having its own library. This library can be used inside the kyma cli for acessing various components. The library contains all the implementation. The kyma cli should invoke this library. We can take inspiration from hydroform and implement it in similar way. Currently we have following modules in kyma:
+
+  1. Lambdas
+  2. Service Catalog
+  3. Events
+  4. Application Connector
+  5. APIs
+
+## Command structure
+
+  The command structure used by cloud providers is following:
+
+  ```bash
+  <cliName> <functionality> <action>
+  #example aws s3 ls
+  ```
+
+  The current kyma-cli also uses the same pattern eg.
+
+  ```bash
+  kyma provision
+  or
+  kyma install
+  ```
+
+### Lambdas
+
+For the serverless component we should start with supporting following commands
+
+#### Local workspace setup
+
+1. Creation of project locally
+  `kyma lambda init <lambda-name> --runtime <runtime-name> -n <namespace>` or `kyma lambda init <lambda-name> --runtime <runtime-name> -n <namespace> -p <path>`
+  This should create directoy with following content based on the runtime:
+
+      ```bash
+        <lambda-name>
+        ├── .vscode
+        │   └── launch.json
+        ├── deployment
+        │   └── deployment.yaml
+        ├── local
+        │   └── index.js
+        └── code
+          ├── config.yaml
+          ├── handler.js
+          └── package.json
+      ```
+
+    > When path `-p` is passed then it should create in desitnation path else it use the workspace set in the yaml file.
+
+    * launch.json consists of settings for the debugging
+    * deployment.yaml consists of the yaml that has been deployed on the kyma cluster.
+    * `index.js` consists of the server code with handler pointing to `handler.js`
+    * config.yaml can be used as info file with current configurations like:
+      - name
```

Comment:
So the config.yaml mainly contains all properties of the function resource, so that the user do not need to understand the k8s specific way of expressing things, right? 
Can you add a discussion about pro and cons and what it would mean to use a function resource file instead which can be deployed also without kymaCLI. I think that discussion we will have a lot later so we should have an arguing.

Reply:
updated.